src/Payum/Core/CoreGatewayFactory.php

```diff
@@ -59,10 +62,21 @@ public function createConfig(array $config = array())
         $config['twig.env'] = null;
 
         $config->defaults(array(
-            'payum.template.layout' => '@PayumCore/layout.html.twig',
+            'httplug.client'=>function(ArrayObject $config) {
+                return HttpClientDiscovery::find();
+            },
+            'httplug.message_factory'=>function(ArrayObject $config) {
+                return MessageFactoryDiscovery::find();
+            },
+            'payum.http_client'=>function(ArrayObject $config) {
+                $func = $config['httplug.client'];
+                if (is_callable($func)) {
+                    $func = $func($config);
+                }
 
-            'payum.http_client' => HttpClientFactory::create(),
-            'guzzle.client' => HttpClientFactory::createGuzzle(),
```

Comment:
Could we add this back `guzzle.client` with a closure which throws an exception with a message why it does not exist any more.